zebra/zebra_dplane.c

```diff
@@ -2853,9 +2853,18 @@ int dplane_ctx_route_init(struct zebra_dplane_ctx *ctx, enum dplane_op_e op,
 	const struct interface *ifp;
 	struct dplane_intf_extra *if_extra;
 
-	if (!ctx || !rn || !re)
+	if (!ctx)
 		return ret;
 
+	/* This function may be called to create/init a dplane context, not
+	 * necessarily to copy a route object.
+	 */
+	if (!rn || !re) {
+		dplane_intf_extra_list_init(&ctx->u.rinfo.intf_extra_list);
```

Comment:
maybe it would make sense to make `route_init_basic()` able to do these key internal inits but then return early if its 're' is NULL? I'm just a little concerned that having two places is going to leave us with some other un-init problem in the future.

Reply:
@mjstapp Make sense. Thanks for the comment. I updated the PR, can you take a look?